plugins/plugin-java-test-runner/che-java-testing-core/che-java-testing-core-ide/src/main/java/org/eclipse/che/ide/ext/java/testing/core/client/view/TestResultViewImpl.java

```diff
@@ -0,0 +1,224 @@
+                return String.valueOf(item.hashCode());
+            }
+        };
+
+        NodeStorage nodeStorage = new NodeStorage(idProvider);
+        NodeLoader nodeLoader = new NodeLoader(Collections.<NodeInterceptor>emptySet());
+
+        resultTree = new Tree(nodeStorage, nodeLoader);
+
+        resultTree.getSelectionModel().addSelectionHandler(new SelectionHandler<Node>() {
+            @Override
+            public void onSelection(SelectionEvent<Node> event) {
+                Node methodNode = event.getSelectedItem();
+                if (methodNode instanceof TestResultMethodNode) {
+                    outputResult.setText(((TestResultMethodNode) methodNode).getStackTrace());
+                }
+                //Log.info(TestResultViewImpl.class, event.getSelectedItem().getName());
+            }
+        });
+
+        resultTree.getElement().getStyle().setWidth(100, Style.Unit.PCT);
+        resultTree.getElement().getStyle().setHeight(100, Style.Unit.PCT);
+        navigationPanel.add(resultTree);
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void focusView() {
+//        tree.setFocus(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void showResults(TestResult result) {
+
+        this.lastTestResult = result;
+        setTitle("Test Results (Framework: " + result.getTestFramework() + ")");
+
+        buildTree();
+
+        focusView();
+    }
+
+    private void buildTree() {
+
+        resultTree.getNodeStorage().clear();
+        outputResult.setText("");
+
+        TestResultGroupNode root = nodeFactory.getTestResultGroupNode(lastTestResult);
+
+        HashMap<String, List<Node>> classNodeHashMap = new HashMap<>();
+
+        for (Failure failure : lastTestResult.getFailures()) {
+            if (!classNodeHashMap.containsKey(failure.getFailingClass())) {
+                List<Node> methodNodes = new ArrayList<>();
+                classNodeHashMap.put(failure.getFailingClass(), methodNodes);
+            }
+            classNodeHashMap.get(failure.getFailingClass())
+                    .add(nodeFactory.getTestResultMethodNodeNode(failure.getFailingMethod(), failure.getTrace(),
+                            failure.getMessage(), failure.getFailingLine(), this));
+        }
+
+        List<Node> classNodes = new ArrayList<>();
+        for (Map.Entry<String, List<Node>> entry : classNodeHashMap.entrySet()) {
+            TestResultClassNode classNode = nodeFactory.getTestResultClassNodeNode(entry.getKey());
+            classNode.setChildren(entry.getValue());
+            classNodes.add(classNode);
+        }
+
+        root.setChildren(classNodes);
+
+        resultTree.getNodeStorage().add(root);
+
+    }
+
+    @Override
+    public void gotoClass(String packagePath, int line) {
+
+        lastWentLine = line;
+
+        final Project project = appContext.getRootProject();
+
+
+        String testSrcPath = project.getPath() + "/src/test/java/";
```

Comment:
I think not good hardcode here path to tests classes. We can use some attributes from projecttype